src/filters/oidc/jwks_resolver.cc

```diff
@@ -37,8 +40,19 @@ void DynamicJwksResolverImpl::JwksFetcher::request(
       }
     }
 
+    if (parent_->jwks() == nullptr) {
+      // On Kubernetes, depending on the timing of the Pod startup,
+      // the first egress HTTP/HTTPS request may fail. This problem
+      // goes away after a few seconds from the start of the pod,
+      // but by default, the timer is rescheduled after 20 minutes,
+      // so if the first fetch fails, the JWKs will be empty for 20 minutes.
+      // To prevent this problem, the timer should be rescheduled after 3
+      // seconds if the acquisition of JWKs fails.
+      next_schedule_interval = std::chrono::seconds(3);
```

Comment:
this this makes sure that we always retry every 3 seconds util the first jwks is successsfully initialized, is that correct?

Reply:
Yes. Once JWKs is set, it will never be set to true. This means that it will only be called if a connection error or HTTP error occurs and JWKs are not set from the system startup.